### PR TITLE
Introduce pkg64cb for the benchmark ABI

### PR DESCRIPTION
**Pull request by kwitaszczyk**
This PR adds `pkg64cb` and other relevant changes to allow building packages with Poudriere and using them with the base system compiled for CheriABI.

An example package repository for `CheriBSD:20230804:aarch64cb` with ports-mgmt/pkg, archivers/zstd and its dependencies is available at pkg.CheriBSD.org. Once this PR is accepted, I will also merge changes in other repositories:
* https://github.com/CTSRD-CHERI/pkg/tree/1.20.5-cheriabi-benchmarkabi
* https://github.com/CTSRD-CHERI/cheribsd-ports/tree/benchmarkabi
* https://github.com/CTSRD-CHERI/poudriere/tree/benchmarkabi

### CODE REVIEW

File: etc/mtree/BSD.pkg64cb.dist [hunk 1]
```diff
@@ -0,0 +1,16 @@
+# $FreeBSD$
```
Comment (brooksdavis) [inline etc/mtree/BSD.pkg64cb.dist:1]:
```suggestion
```

**Review by brooksdavis — COMMENTED** [review]
The first commit message should likely provide some justification for the ordering.

**Comment by kwitaszczyk** [comment]
> The first commit message should likely provide some justification for the ordering.

I updated the commit message.

**Review by brooksdavis — APPROVED** [review]

**Review by jrtc27 — COMMENTED** [review]
Just a couple of comments about clarifying wording in pkg.sh

File: usr.sbin/pkg64c/pkg.sh [hunk 2]
```diff
@@ -31,9 +31,10 @@
 cat >&2 <<EOF
 Please use the ABI-specific pkg variant:
  * pkg64c for experimental software built for CheriABI;
```
[the remark this answers is not in the document]
Comment (kwitaszczyk) [reply usr.sbin/pkg64c/pkg.sh]: That's a good idea. What about "for the pure-capability ABI (CheriABI)" (the current version)?

**Comment by kwitaszczyk** [comment]
Let's merge it now, and we can improve it separately, if needed.

**Pull request merged**